public/components/map_top_nav/top_nav_menu.tsx

```diff
@@ -90,13 +105,17 @@ export const MapTopNavMenu = ({
   };
 
   useEffect(() => {
-    setDateFrom(mapState.timeRange.from);
-    setDateTo(mapState.timeRange.to);
+    if (!inDashboardMode) {
+      setDateFrom(mapState.timeRange.from);
+      setDateTo(mapState.timeRange.to);
+    } else {
+      setDateFrom(timeRange!.from);
+      setDateTo(timeRange!.to);
+    }
     setQueryConfig(mapState.query);
     setIsRefreshPaused(mapState.refreshInterval.pause);
     setRefreshIntervalValue(mapState.refreshInterval.value);
-    refreshDataLayerRender();
-  }, [mapState]);
+  }, [mapState, timeRange]);
```

Comment:
why we remove `refreshDataLayerRender`? where does refreshDataLayerRender been called?

Reply:
The same effect will be take at https://github.com/opensearch-project/dashboards-maps/pull/231/files#diff-48b7512b7ac9706ea9d00115c9b8aee0119c926eec3f18387a2d443ddd92d23bR225. Finally we plan to move all the data handler into one place.